File: sdk/scala-finch/src/main/scala/org/openapitools/models/Item.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import java.math.BigDecimal
import scala.collection.immutable.Seq

/**
 * 
 * @param upc 
 * @param ean 
 * @param isbn 
 * @param asin 
 * @param title 
 * @param sku 
 * @param mpn 
 * @param partUnderscorenumber 
 * @param upcs A list of matching upc codes.
 * @param description 
 * @param brand 
 * @param manufacturer 
 * @param color 
 * @param newUnderscoreprice 
 * @param usedUnderscoreprice 
 * @param currencyUnderscorecode 
 * @param url 
 * @param features A list of product feature descriptions.
 * @param dimensions Product dimensions and weight.  ('meters', 'centimeters', 'millimeters', 'inches', 'feet', 'yards', 'ounces', 'pounds', 'grams', 'milligrams', 'kilograms')
 * @param images A list product images.
 * @param matchedUnderscoreitems Matched items for this product.
 * @param isoUnderscorecountryUnderscorecodes A list of country codes from which this item belongs.
 * @param companyUnderscorename The company name registered to the product's UPC or EAN.
 * @param companyUnderscoreaddress The company address registered to the product's UPC or EAN.
 * @param categories A list of categories for this product.
 * @param categoryUnderscorehierarchies A list of category hierarchies for this product. Categories are listed in order of generic category to most specific category.
 */
case class Item(upc: Option[String],
                ean: Option[String],
                isbn: Option[String],
                asin: Option[String],
                title: Option[String],
                sku: Option[String],
                mpn: Option[String],
                partUnderscorenumber: Option[String],
                upcs: Option[Seq[String]],
                description: Option[String],
                brand: Option[String],
                manufacturer: Option[String],
                color: Option[String],
                newUnderscoreprice: Option[BigDecimal],
                usedUnderscoreprice: Option[BigDecimal],
                currencyUnderscorecode: Option[String],
                url: Option[String],
                features: Option[Seq[String]],
                dimensions: Option[Seq[Object]],
                images: Option[Seq[String]],
                matchedUnderscoreitems: Option[Seq[Object]],
                isoUnderscorecountryUnderscorecodes: Option[Seq[String]],
                companyUnderscorename: Option[String],
                companyUnderscoreaddress: Option[String],
                categories: Option[Seq[String]],
                categoryUnderscorehierarchies: Option[Seq[String]]
                )

object Item {
    /**
     * Creates the codec for converting Item from and to JSON.
     */
    implicit val decoder: Decoder[Item] = deriveDecoder
    implicit val encoder: ObjectEncoder[Item] = deriveEncoder
}

File: sdk/fsharp-functions/OpenAPI/src/model/Item.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json

module Item = 

  //#region Item

  [<CLIMutable>]
  type Item = {
    [<JsonProperty(PropertyName = "upc")>]
    Upc : string;
    [<JsonProperty(PropertyName = "ean")>]
    Ean : string;
    [<JsonProperty(PropertyName = "isbn")>]
    Isbn : string;
    [<JsonProperty(PropertyName = "asin")>]
    Asin : string;
    [<JsonProperty(PropertyName = "title")>]
    Title : string;
    [<JsonProperty(PropertyName = "sku")>]
    Sku : string;
    [<JsonProperty(PropertyName = "mpn")>]
    Mpn : string;
    [<JsonProperty(PropertyName = "part_number")>]
    PartNumber : string;
    [<JsonProperty(PropertyName = "upcs")>]
    Upcs : string[];
    [<JsonProperty(PropertyName = "description")>]
    Description : string;
    [<JsonProperty(PropertyName = "brand")>]
    Brand : string;
    [<JsonProperty(PropertyName = "manufacturer")>]
    Manufacturer : string;
    [<JsonProperty(PropertyName = "color")>]
    Color : string;
    [<JsonProperty(PropertyName = "new_price")>]
    NewPrice : decimal;
    [<JsonProperty(PropertyName = "used_price")>]
    UsedPrice : decimal;
    [<JsonProperty(PropertyName = "currency_code")>]
    CurrencyCode : string;
    [<JsonProperty(PropertyName = "url")>]
    Url : string;
    [<JsonProperty(PropertyName = "features")>]
    Features : string[];
    [<JsonProperty(PropertyName = "dimensions")>]
    Dimensions : obj[];
    [<JsonProperty(PropertyName = "images")>]
    Images : string[];
    [<JsonProperty(PropertyName = "matched_items")>]
    MatchedItems : obj[];
    [<JsonProperty(PropertyName = "iso_country_codes")>]
    IsoCountryCodes : string[];
    [<JsonProperty(PropertyName = "company_name")>]
    CompanyName : string;
    [<JsonProperty(PropertyName = "company_address")>]
    CompanyAddress : string;
    [<JsonProperty(PropertyName = "categories")>]
    Categories : string[];
    [<JsonProperty(PropertyName = "category_hierarchies")>]
    CategoryHierarchies : string[];
  }
  
  //#endregion
  